src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -1071,6 +1081,40 @@ def _prepare_workspace(cmd, resource_group_name, workspace):
     else:
         workspace_id = workspace
     return workspace_id
+
+
+def _set_data_source_for_workspace(cmd, os_type, resource_group_name, workspace_id):
+    from ._client_factory import cf_log_analytics_data_sources
+    from azure.cli.core.commands.client_factory import get_subscription_id
+    from azure.mgmt.loganalytics.models import DataSource
+    from msrestazure.tools import parse_resource_id
+
+    subscription_id = get_subscription_id(cmd.cli_ctx)
+    data_sources_client = cf_log_analytics_data_sources(cmd.cli_ctx, subscription_id)
+    workspace_name = parse_resource_id(workspace_id)['name']
+    data_source_name_template = "DataSource_{}_{}"
+
+    default_data_sources = None
+    if os_type.lower() == 'linux':
+        from ._workspace_data_source_settings import default_linux_data_sources
+        default_data_sources = default_linux_data_sources
+    elif os_type.lower() == 'windows':
+        from ._workspace_data_source_settings import default_windows_data_sources
+        default_data_sources = default_windows_data_sources
+
+    if default_data_sources is not None:
+        for data_source_kind, data_source_settings in default_data_sources.items():
+            for data_source_setting in data_source_settings:
+                data_source = DataSource(kind=data_source_kind,
+                                         properties=data_source_setting)
+                data_source_name = data_source_name_template.format(data_source_kind, _gen_guid())
+                data_sources_client.create_or_update(resource_group_name,
```

Comment:
What will happen if this operation fails? Will user know it?

Reply:
It will fail. Do you mean I should add try/except here?

Reply:
I think both ways are OK. We need to define when the command execution should be considered successful.